### PR TITLE
support warmup groups in from_pymc3

### LINKED ISSUE #1146: Automatically handle warmup draws and sampling metadata from_pymc3
## Tell us about it

The `trace.report` in PyMC3 now has attributes `n_tune`, `n_draws` and `t_sampling` (see https://github.com/pymc-devs/pymc3/pull/3827).

I have this snippet to store those metadata with the trace:

```
idat = arviz.from_pymc3(trace)
if hasattr(trace.report, 'n_tune'):
        idat.posterior.attrs['n_tune'] = trace.report.n_tune
        idat.posterior.attrs['n_draws'] = trace.report.n_draws
        idat.posterior.attrs['t_sampling'] = trace.report.t_sampling
```

Using `n_draws`, one can now slice a trace that contains warmup iterations:

```
trace_warmup = trace[:-trace.report.n_draws]   # <-- may result in len(trace_warmup) == 0
trace_posterior = trace[-trace.report.n_draws:]
```

## Thoughts on implementation

I suppose both snippets should find their place in the [PyMC3Converter](https://github.com/arviz-devs/arviz/blob/68406c4f86d6e560e948c1dd37c6c4314cc895dd/arviz/data/io_pymc3.py#L43), but I'm not sure how saving the warmup should be done.

### PR DESCRIPTION
**Pull request by OriolAbril**
## Description
Support warmup groups in `from_pymc3`. It should already work, but I have not figured out the dependencies of the whole thing (it may only work on pymc3 master) nor added any test yet. 

Will fix #1146 

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] New features are properly documented (with an example if appropriate)?
- [x] Includes new or updated tests to cover the new feature
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by michaelosthege** [comment]
looks like pylint doesn't want us to access `_warmup_posterior` in the test (protected member)
https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=2376&view=logs&j=e6a7683b-6131-58a8-ef68-5f3a9120796c&t=b52e026f-cc99-5e74-288f-e4d4024efc71&l=15

**Comment by OriolAbril** [comment]
Ready for review or even merge

Also related to #1083, this starts storing number of tuning steps in inference data as attrs of posterior and sample stats (both samples and warmup).

File: arviz/tests/external_tests/test_data_pymc.py [hunk 1]
```diff
@@ -397,3 +397,36 @@ def test_no_model_deprecation(self):
...
+            "~log_likelihood": [],
+        }
+        fails = check_multiple_attrs(test_dict, idata)
+        assert not fails
+        assert idata.posterior.sizes["chain"] == 2
+        assert idata.posterior.sizes["draw"] == 200
+        if save_warmup:
+            # pylint: disable=protected-access
+            assert idata._warmup_posterior.dims["chain"] == 2
+            assert idata._warmup_posterior.dims["draw"] == 100
```
Comment (AlexAndorra) [inline arviz/tests/external_tests/test_data_pymc.py:432]: - Just out of curiosity: why is the attribute to access different for `posterior` (`sizes`) and `_warmup_posterior` (`dims`), whereas you want to see the number of chains and draws in both cases?
- Also, does it mean we keep the `_` before `warmup`?

Comment (ahartikainen) [reply arviz/tests/external_tests/test_data_pymc.py:432]: _ should stay before warmup. it is not a mcmc draw (in hmc; maybe) so it is only for advanced users

Comment (OriolAbril) [reply arviz/tests/external_tests/test_data_pymc.py:432]: Regarding first question, both attributes do the same, the test Michael sent had sizes and I used dims to make it more xarray-y but missed half of them :shrug:

Comment (OriolAbril) [reply arviz/tests/external_tests/test_data_pymc.py:432]: @ahartikainen how opposed would you be to removing the `_` before warmup? Even if only for advanced users it is still for users, the default would still be `data.save_warmup = false` so unless they change it they would not have warmup groups, I feel like hiding them so much makes harder the live of advanced users without much gain for novices.

Comment (ahartikainen) [reply arviz/tests/external_tests/test_data_pymc.py:432]: Not too opposed.

**Comment by OriolAbril** [comment]
I think it is ready to merge

**Comment by AlexAndorra** [comment]
Agreed 👍

**Pull request merged**